p2p/protocol/identify/obsaddr.go

```diff
@@ -534,6 +527,17 @@ func (oas *ObservedAddrManager) emitSpecificNATType(addrs []*observedAddr, proto
 	return false, 0
 }
 
+func (oas *ObservedAddrManager) Close() error {
+	oas.closeOnce.Do(func() {
+		oas.ctxCancel()
+		oas.refCount.Wait()
+		oas.reachabilitySub.Close()
+		oas.refreshTimer.Stop()
```

Comment:
1. I believe we still need to hold the lock here because `SetTTL` is a public function.
2. We should _probably_ nil out the `refreshTimer`, and check if it's nil in `SetTTL`.

Reply:
I was assuming that you only call `Close` when you're done using the struct, and don't concurrently call `Close` with other methods.
On the other hand, locking the mutex here is cheap, so why not...

Reply:
Ideally, but it can be hard to really guarantee that.